test/hotspot/jtreg/serviceability/attach/LongArgTest.java

```diff
@@ -0,0 +1,161 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @summary Tests that long arguments of attach operation are not truncated
+ * @bug 8334168
+ * @library /test/lib
+ * @modules jdk.attach/sun.tools.attach
+ * @run main LongArgTest
+ */
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.InputStream;
+import java.io.IOException;
+
+import com.sun.tools.attach.VirtualMachine;
+import sun.tools.attach.HotSpotVirtualMachine;
+
+import jdk.test.lib.apps.LingeredApp;
+
+public class LongArgTest {
+
+    public static void main(String[] args) throws Exception {
+        LingeredApp app = null;
+        try {
+            app = LingeredApp.startApp();
+
+            test(app).run();
+
+            test(app)
+                .withLongValue()
+                .run();
+
+            test(app)
+                .withSuperLongValue()
+                .run();
+
+        } finally {
+            LingeredApp.stopApp(app);
+        }
+
+    }
+
+    // For simplicity, the test uses internal HotSpotVirtualMachine,
+    // sets/gets "HeapDumpPath" flag value (string flag, not validated by JVM).
+    private static class Test {
+        private LingeredApp app;
+        private String flagName = "HeapDumpPath";
+        private String flagValue = generateValue(5);
+
+        Test(LingeredApp app) {
+            this.app = app;
+        }
+
+        // Value length exceeds 1K.
+        Test withLongValue() {
+            flagValue = generateValue(1024 + 1);
+            return this;
+        }
+
+        // Value length exceeds 3K (1K * 3 args).
+        Test withSuperLongValue() {
+            flagValue = generateValue(3 * 1024 + 1);
+            return this;
+        }
+
+        void run() throws Exception {
+            System.out.println("======== Start ========");
+
+            HotSpotVirtualMachine vm = (HotSpotVirtualMachine)VirtualMachine.attach(String.valueOf(app.getPid()));
+
+            if (setFlag(vm)) {
```

Comment:
Currently Attach operations have restriction for argument size, so setFlag() is expected to fail for long values.
This fix adds AttachOperationFailedException for the case on windows, linux/bsd/aix implementations throw generic IOException.
(There is [JDK-8334168](https://bugs.openjdk.org/browse/JDK-8334168) to throw AttachOperationFailedException instead of IOException if an argument is too long on other platforms)

Reply:
Not sure, you've answered my question. Let me ask it differently.
Here I do not care about other exceptions on other platforms.
The `setFlag()` can catch an `IOException` at the line 126 and return false in that case. The `run()` method at the line 94 is swallowing the `false` value as there is no `else` statement. So, we just ignore the `IOException` and do not test anything.
My question is: Why don't we throw a `RuntimeException` in the case the `setFlag()` returned `false`?